src/mono/wasm/build/WasmApp.Common.targets

```diff
@@ -156,8 +156,8 @@
 
     <WasmDedup Condition="'$(WasmDedup)' == ''">true</WasmDedup>
     <_IsLibraryMode Condition="'$(OutputType)' == 'Library' and '$(UsingMicrosoftNETSdkWebAssembly)' == 'true'">true</_IsLibraryMode>
-    <IsWasiProject Condition="'$(IsWasiProject)' == '' and '$(RuntimeIdentifier)' == 'wasi-wasm' and ('$(OutputType)' != 'Library' or '$(_IsLibraryMode)' == 'true')">true</IsWasiProject>
-    <IsBrowserWasmProject Condition="'$(IsBrowserWasmProject)' == '' and '$(RuntimeIdentifier)' == 'browser-wasm' and ('$(OutputType)' != 'Library' or '$(_IsLibraryMode)' == 'true')">true</IsBrowserWasmProject>
+    <IsWasiProject Condition="'$(IsWasiProject)' == '' and '$(RuntimeIdentifier)' == 'wasi-wasm' and '$(OutputType)' != 'Library' and '$(_IsLibraryMode)' != 'true'">true</IsWasiProject>
```

Comment:
`$(_IsLibraryMode) != true` doesn't look right 🤔

Reply:
It should work the same way for browser-wasm, as https://github.com/dotnet/runtime/pull/106388. We are missing a test case for that and that's something we should fix. Treating Wasm SDK libraries as applications in intended for both browser & wasi. As described in the first mentioned PR
- RID (or ideally TFM) should give access to browser/wasi specific APIs
- SDK should access wasm bundling

Do you have a case for using Wasm SDK in project that shouldn't be treated as application?

Reply:
When creating a class library which uses the "Microsoft.NET.SDK.WebAssembly" SDK, building causes the following error:

```
WasmAssembliesToBundle item is empty. No assemblies to process
```

I'll create an issue with it, as it seems that the resolution is a bit more complex than the change I'm suggesting :)

Reply:
Thank you!